sys/include/sema.h

```diff
@@ -191,16 +187,39 @@ static inline int sema_wait(sema_t *sema)
  */
 static inline int sema_try_wait(sema_t *sema)
 {
-#if IS_USED(MODULE_ZTIMER)
     return _sema_wait_ztimer(sema, 0, NULL, 0);
-#else
-    return _sema_wait_xtimer(sema, 0, 0);
-#endif
 }
 
-#if IS_USED(MODULE_XTIMER) || defined(DOXYGEN)
+#if IS_USED(MODULE_SEMA_ZTIMER64) || defined(DOXYGEN)
+/**
+ * @brief   Wait for a semaphore being posted, using ztimer as backend
+ *
+ * @pre `(sema != NULL)`
+ * @pre `(clock != NULL)`
+ *
+ * @param[in]  sema     A semaphore.
+ * @param[in]  clock    ztimer clock to use
+ * @param[in]  timeout  Time in microseconds until the semaphore times out.
+ *                      0 does not wait.
+ *
+ * @return  0 on success
+ * @return  -ETIMEDOUT, if the semaphore times out.
+ * @return  -ECANCELED, if the semaphore was destroyed.
+ * @return  -EAGAIN,    if the semaphore is not posted (only if timeout = 0)
+ */
+static inline int sema_wait_timed_ztimer64(sema_t *sema,
+                                           ztimer64_clock_t *clock,
+                                           uint32_t timeout)
+{
+    return _sema_wait_ztimer64(sema, (timeout != 0), clock, timeout);
+}
+#endif
```

Comment:
i don't think this api should be created or it should be marked internal use only.

(i somehow can not imagine a case where someone wants to wait for a semaphorer for a time longer than 1h but not infinite) 

1h ~ fits into 32Bit of usec or 22Bit of msec

Reply:
(there certainly is the case of posix semaphore , and i found a similar one in wasm but in both cases it seems to me that the person who designed that api just used the time type that allready was there)

Reply:
Ok so drop this from the public API!